core/chains/evm/logpoller/log_poller.go

```diff
@@ -27,7 +27,7 @@ import (
 type LogPoller interface {
 	services.ServiceCtx
 	Replay(ctx context.Context, fromBlock int64) error
-	MergeFilter(topics []common.Hash, address common.Address)
+	MergeFilter(eventSigs []common.Hash, addresses []common.Address)
```

Comment:
Maybe add a godoc here:

```golang
// MergeFilter adds the provided event signatures and addresses to the log poller's log filter query.
// If an event matching any of the given event signatures is emitted from any of the provided addresses,
// the log poller will pick those up and save them.
```

And maybe add a note if someone wants to do a more sophisticated query, they'd have to do a SQL query and match against topic1, topic2, etc. if they'd like more fine-grained event filtering.

Reply:
FWIW I think it would be useful to have a godoc on all of the methods in this interface - one thing that might be worthy of note (that I was not aware of) is that the log poller saves heads to the database, similarly to the head saver. Maybe some note on how the log poller is different from the head saver in that regard (e.g as part of a godoc on the `LatestBlock` method) would help casual onlookers.